accepted/00000-hub-reporting.md

```diff
@@ -0,0 +1,199 @@
+
+![Server schema example](images/hub-reporting/server-example.png)
+
+This data structure can be used to extract the pieces of information that are currently available through the reports:
+
+- inventory
+- inactive systems
+- hosts/guests
+- system profile
+
+This existing reports can be provided in the new schema as views, ready to be consumed. In addition, the user will
+have the possibility to extract the pieces of information he needs by querying the data directly from the tables and
+combining them as needed.
+
+# Installation
+The reporting tools will be installed by default; all the upgrade to a newer version, will install the 
+reporting tools as well. Several packages will be provided in order to provide the following tools and
+functionalities:
+- setup the database and initialize the schema,
+- perform database schema migration,
+- setup database users,
+- setup tasktomatic jobs.
+
+All the other information required (e.g. Uyuni Server info) will be retrieved by already existing configuration.
+
+## Reporting Database
+A reporting DB is setup in all Uyuni Servers and Uyuni Hub nodes to store reporting data.
+The default setup will create a Read/Write user, to manage the database and write the data from it into the Reporting DB.
+This user and the DB connection parameters are written into `/etc/rhn/rhn.conf` similar to the default DB options.
+
+## Initial Setup
+Uyuni Hub should create its own read-only user on the Reporting Databases of the single Servers.
+As the Servers are managed with salt, we will write a state to create an account.
+The username and password are generated on the Hub and provided as pillar data.
+On the server the state take care of the existence of the account and the Hub
+can store the parameters in its database under the system entry.
+
+This account does not need to be stored in plain text on the managed Server and can be kept only on the Hub.
+The same apply for accounts created by the administrator used for Reporting Tools which need access
+to the Reporting Database.
+
+
+# Workflow
+
+## Scheduled job: collect from Uyuni DB
+A taskomatic job is responsible to fetch and prepare the data from the main Uyuni
+Database and insert them into the Reporting Database.
+The job should be written in a way, that the Reporting Database could be on different Host.
+
+Keeping the code which insert the data into the reporting DB in sync with the Reporting Database Schema
+is a requirement.
+
+Implementing a taskomatic simple java job should be sufficient as we need only one task which run at a
+certain point in time.
+
+As we design this feature to be ready to use an external database for reporting, we do not use materialized views.
+
+## Scheduled job: collect remote data
+We also have an additional taskomatic job which collect the data from all the managed
+Uyuni servers and insert them into the Hub Reporting Database which could be
+again an external DB.
```

Comment:
This will be a delta between the last sync time and the current time, a few words about "how" part in this regard, in RFC will be nice.

Reply:
Moreover, it will not always be the case that data will be inserted, it could be also deletion. We have solved a similar issue in ISSv2 so might be helpful to take a look at that approach.

Reply:
The idea was to do a "delete all and re-insert all". I will write this.